### PR TITLE
ipctest: Fix random hang during startup.

### PR DESCRIPTION
**Pull request by kinetiknz**
Observed on macOS: sometimes during startup, ipctest would hang after
printing the startup banner.

It turns out that the ipctest child could hang waiting for a lock deep
inside tokio; however the lock had been taken (and dropped) in the
parent during initialization.  fork() would just happen to clone the
process at a bad time, leaving tokio's internal state unusable in the
child.

To avoid this entire class of problem, ipctest on Unix now uses a
similar technique to Windows to bootstrap - after the client fd is
allocated, ipctest forks and then execs to reset its process state
before continuing with the audioipc tests.

r? @ChunMinChang please

### CODE REVIEW

File: ipctest/src/main.rs [hunk 1]
```diff
@@ -27,14 +27,34 @@ use crate::errors::*;
...
+
     let handle =
         unsafe { audioipc_server::audioipc_server_start(std::ptr::null(), std::ptr::null()) };
     let fd = audioipc_server::audioipc_server_new_client(handle);
+    let fd = unsafe {
+        let new_fd = libc::dup(fd);
+        libc::close(fd);
+        new_fd
+    };
+    assert!(fd >= 0);
```
Comment (ChunMinChang) [inline ipctest/src/main.rs]: `fd > INVALID_HANDLE_VALUE` might work since `INVALID_HANDLE_VALUE` is a `pub const` now?

File: ipctest/src/main.rs [hunk 2]
```diff
@@ -27,14 +27,34 @@ use crate::errors::*;
...
 
     match unsafe { libc::fork() } {
         -1 => bail!("fork() failed"),
         0 => {
-            return client::client_test(fd);
+            let self_path = CString::new(&*args[0]).unwrap();
+            let child_arg1 = CString::new("--client").unwrap();
+            let child_arg2 = CString::new("--fd").unwrap();
+            let child_arg3 = CString::new(format!("{}", fd)).unwrap();
+            let child_args = [self_path.as_c_str().as_ptr(),
```
Comment (ChunMinChang) [inline ipctest/src/main.rs]: `self_path.as_ptr()` could work. It seems [`CString::as_ptr()`](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr) is a [shortcut for `CString::as_c_str()::as_ptr()`](https://doc.rust-lang.org/std/ffi/struct.CString.html#extracting-a-raw-pointer-to-the-whole-c-string).

Comment (kinetiknz) [reply ipctest/src/main.rs]: That's tidier, thanks!

**Comment by kinetiknz** [comment]
Oops, I accidentally merged this without the review comments addressed.  I'll push those in a follow up now.

**Pull request merged**